pkg/block/blocktest/adapter.go

```diff
@@ -463,3 +342,62 @@ func testAdapterWalker(t *testing.T, adapter block.Adapter, storageNamespace str
 		})
 	}
 }
+
+// Test request for a presigned URL for temporary access
+func testGetPreSignedURL(t *testing.T, adapter block.Adapter, storageNamespace string) {
+	ctx := context.Background()
+	obj, _ := objPointers(storageNamespace)
+
+	preSignedURL, exp, err := adapter.GetPreSignedURL(ctx, obj, block.PreSignModeRead)
+
+	if adapter.BlockstoreType() == block.BlockstoreTypeGS {
+		require.ErrorContains(t, err, "no credentials found")
+		return
+	} else if adapter.BlockstoreType() == block.BlockstoreTypeLocal {
+		require.ErrorContains(t, err, "operation not supported")
+		return
+	}
+	require.NoError(t, err)
+	expectedExpiry := expectedURLExp(adapter)
+	require.Equal(t, expectedExpiry, exp)
+	_, err = url.Parse(preSignedURL)
+	require.NoError(t, err)
+}
+
+func expectedURLExp(adapter block.Adapter) time.Time {
+	if adapter.BlockstoreType() == block.BlockstoreTypeAzure {
+		// we didn't implement expiry for Azure yet
+		return time.Time{}
+	} else {
+		return NowMockDefault().Add(block.DefaultPreSignExpiryDuration)
+	}
+}
+
+func dumpPathTree(t testing.TB, ctx context.Context, adapter block.Adapter, qk block.QualifiedKey) []string {
+	t.Helper()
+	tree := make([]string, 0)
+
+	uri, err := url.Parse(qk.Format())
+	require.NoError(t, err)
+
+	w, err := adapter.GetWalker(uri)
+	require.NoError(t, err)
+
+	walker := store.NewWrapper(w, uri)
+	require.NoError(t, err)
+
+	err = walker.Walk(ctx, block.WalkOptions{}, func(e block.ObjectStoreEntry) error {
+		_, p, _ := strings.Cut(e.Address, uri.String())
+		tree = append(tree, p)
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("walking on '%s': %s", uri.String(), err)
```

Comment:
This is a bit of an issue. On the one hand, calling `t.Helper()` in this function means the failing test result will point you to the call site - line 167. This is usually what you want from a helper function, especially if it's used in multiple tests/test cases.

However, since this function contains multiple assertions, it could be difficult to tell which one failed.  
`require.NoError` has a third argument, allowing you to add a custom message. The message will be part of the reported test output. This can help you pinpoint the exact place where the error occurred.